### PR TITLE
add MonadFail instances for Conversion and RowParser

### PR DESCRIPTION
**Pull request by chris-martin**
I'm trying to update some code to GHC 8.8, and I've got a handful of these errors:

```
    • No instance for (MonadFail RowParser)
        arising from a use of ‘fail’
```

I believe this is a pretty straightforward update, unless there is some reason you do not want RowParser to have `fail` anymore?

### CODE REVIEW

File: src/Database/PostgreSQL/Simple/Internal.hs [hunk 1]
```diff
@@ -497,6 +498,9 @@ instance MonadPlus Conversion where
    mzero = empty
    mplus = (<|>)
 
+instance Fail.MonadFail Conversion where
```
Comment (phadej) [inline src/Database/PostgreSQL/Simple/Internal.hs]: `Monad Conversion` doesn't have `fail`. There should be properly guarded

```
   fail = MonadFail.fail
```

**Comment by chris-martin** [comment]
> As `default fail = error`

Oh, I see, I was mistaken. So I suppose I don't really need this change after all, I could just switch to `error`.

But would you like me to make these changes and make a proper `MonadFail` instance anyway?

**Comment by phadej** [comment]
I think people should use `conversionError` with proper exception type. I don't think we should promote `MonadFail` here.

**Comment by chris-martin** [comment]
`conversionError` gives me a `Conversion`; how do I convert that to a `RowParser`?

**Comment by phadej** [comment]
If you definitely don't want `FieldParser` (which is a type synonym with `-> Conversion a` at the end), then you can use internals https://hackage.haskell.org/package/postgresql-simple-0.6.2/docs/Database-PostgreSQL-Simple-Internal.html

Most likely you want to use `FieldParser`.

**Comment by chris-martin** [comment]
Okay, thanks.

**Pull request closed without merging**